trin-core/src/types/accumulator.rs

```diff
@@ -308,6 +232,12 @@ pub struct HeaderRecord {
     pub total_difficulty: U256,
 }
 
+fn calculate_generalized_index(header: &Header) -> usize {
+    // Calculate generalized index for header
+    let hr_index = header.number as usize % EPOCH_SIZE;
+    (EPOCH_SIZE * 2 * 2) + (hr_index * 2)
```

Comment:
It's a bit unclear what the calculation is here. 

```rs
(EPOCH_SIZE * 2 * 2) + (hr_index * 2)
```

Otherwise, all looks good to me. Tested locally, no issues found.